### PR TITLE
Refactor transcription line to finish on pointerup

### PR DESCRIPTION
**Pull request by mcbouslog**
Package: lib-classifier

Towards #1546 (Item 1)

Describe your changes:
- refactor transcription line to use component state to track pointer events, finish on second pointerup

Potentially helpful staging workflow:
- http://localhost:8080/?project=1764&workflow=3392

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?

## Components
- [ ] Has a storybook story been created or updated?
- [ ] Is the component accessible? 
  - [ ] Can it be used with a screen reader? [BBC guide to testing with VoiceOver](https://bbc.github.io/accessibility-news-and-you/accessibility-and-testing-with-voiceover-os.html)
  - [ ] Can it be used from the keyboard? [WebAIM guide to keyboard testing](https://webaim.org/techniques/keyboard/#testing)
  - [ ] Is it passing accessibility checks in the storybook?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `yarn panic && yarn bootstrap` or `docker-compose up --build` and app works as expected?

## Publishing

- [ ] Is the changelog updated?
- [ ] Are the dependencies updated for apps and libraries that are using the newly published library?

## Post-merging

- [ ] Did the app deploy to https://frontend.preview.zooniverse.org/projects/:project-name/:owner or https://frontend.preview.zooniverse.org/about?
- [ ] Is the new feature working or bug now fixed?
  - [ ] Is there a Talk or blog post written to announce the new feature(s)?
- [ ] Is the design working across browsers (Firefox, Chrome, Edge, Safari) and mobile?
  - [ ] Is this approved by our designer?
- [ ] Is this ready for production deployment?

### CODE REVIEW

**Comment by mcbouslog** [comment]
Focusing on second click/point of transcription line's pointerup and pointerdown...

Current (including #1677) UI:
pointerDownSubtask

These changes:
movePointThenSubtask

File: packages/lib-classifier/src/plugins/drawingTools/experimental/components/TranscriptionLine/TranscriptionLine.spec.js [hunk 1]
```diff
@@ -227,8 +227,43 @@ describe('Components > Drawing marks > Transcription line', function () {
         }
       )
       const dragMove = wrapper.find(DragHandle).find('[x=300]').at(0).prop('dragMove')
+      
+      wrapper.find('circle[cx=300]').at(0).simulate('pointerdown')
+      expect(mark.x2).to.equal(300)
+      expect(mark.y2).to.equal(400)
+      dragMove({}, { x: 10, y: 20 })
```
Comment (mcbouslog) [inline packages/lib-classifier/src/plugins/drawingTools/experimental/components/TranscriptionLine/TranscriptionLine.spec.js:234]: I initially thought this should be something along the lines of:
`wrapper.find('circle[cx=300]').at(0).simulate('pointermove', { x: 10, y: 20 })`
but was unable successfully implement.

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/DrawingToolMarks/DrawingToolMarks.js [hunk 2]
```diff
@@ -57,6 +57,14 @@ function DrawingToolMarks ({ activeMarkId, marks, onDelete, onDeselectMark, onFi
       }
     }
 
+    function endMoveMark (event) {
```
Comment (srallen) [inline packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/DrawingToolMarks/DrawingToolMarks.js:60]: This seems to be a duplicate of lines 52-58. Rebase mistake?

**Comment by srallen** [comment]
Does this close https://github.com/zooniverse/front-end-monorepo/issues/1640 too?

File: packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/DrawingToolMarks/DrawingToolMarks.js (hunk 2 above)
Comment (mcbouslog) [reply packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/InteractionLayer/components/DrawingToolMarks/DrawingToolMarks.js]: It was indeed a rebase mistake, removed.

**Comment by mcbouslog** [comment]
This fixes the issue with the subtask automatically opening on pointerdown (as noted in this [1640 comment](https://github.com/zooniverse/front-end-monorepo/issues/1640#issuecomment-634228980)), however, it doesn't fix the two touch expectation noted in #1640's original comment, and the subtask doesn't open unless the transcription line is touched again (selected) after creation. 

The touch behavior with this PR is that a transcription line is created with pointer/touch/down, then continue touch to drag and create line, then on pointer/touch/up it finishes creating the line (one touch event), which is not the desired two touch events.

**Pull request merged**